app/controllers/users_controller.rb

```diff
@@ -0,0 +1,36 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show, :friends]
+  before_action :set_current_user, only: [:me, :edit]
+
+  def index
+    friends               = current_user.friends.ids
```

Comment:
Lines 6-9 probably are giving too much burden on this index action. At its core the users index action should be only concerned with showing 'users' (filtered depending on the scope you used in the model). Then all others like getting their friends, etc might be better done in another controller or even possibly in the model itself and just invoked here.

Reply:
I moved logic to model, if you have a better idea, I'd like to know it